datajunction-server/datajunction_server/construction/build.py

```diff
@@ -489,6 +489,10 @@ def build_node(  # pylint: disable=too-many-arguments
             ),
         )
 
+    # get dimension columns which are required
```

Comment:
Does this need to additionally check if the user-provided dimensions overlap with the `bound_dimensions` set on the node?